whois-client/src/main/java/net/ripe/db/whois/common/sso/SsoTranslation.java

```diff
@@ -0,0 +1,52 @@
+package net.ripe.db.whois.common.sso;
+
+
+import com.google.common.collect.Maps;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+
+public class SsoTranslation {
+
+    private final Map<String, String> usernameToUuidCache = Maps.newHashMap();
+
+    public void put(final String username, final String uuid) {
+        if (!containsUsername(username) && !containsUuid(uuid)) {
```

Comment:
Correct me if I am wrong, This seems to me some kind of cache for uuid against username and vice-versa. If that is the case Authservice client is already using cache so why not directly call the Authservice client instead of maintaining another cache ?

Reply:
you are right, we are caching authservices
yes, this map of username and uuid is like a cache. However, I think that the main purpose is to detect whenever there is a duplicated SSO.
Example: 
auth: SSO a@ripe.net //uuid = 123
auth: SSO 123 //a@ripe.net

However, we don't really care about this, because if we receive an uuid we will thrown an exception. So I think that the main idea of using this is to detect this behaviour and to send (in the code of the old branch, was an error...which was breaking the syncupdates, returning 500...I updated that to send a warn instead...however in the standup you told me that we just want to not allow directly the UUID so return an ***Error in a synupdate response and 400 in the rest call)

So you are right but I think that the purpose of this is to cache and to avoid duplications between UUID and Email